cmd/kubeadm/app/util/endpoint.go

```diff
@@ -36,16 +37,26 @@ func GetMasterEndpoint(cfg *kubeadmapi.MasterConfiguration) (string, error) {
 	return fmt.Sprintf("https://%s", hostPort), nil
 }
 
-// GetMasterHostPort returns a properly formatted Master IP/port pair or error
-// if the IP address can not be parsed or port is outside the valid TCP range.
+// GetMasterHostPort returns a properly formatted Master hostname or IP and port pair, or error
+// if the hostname or IP address can not be parsed or port is outside the valid TCP range.
 func GetMasterHostPort(cfg *kubeadmapi.MasterConfiguration) (string, error) {
 	var masterIP string
+	var portStr string
 	if len(cfg.API.ControlPlaneEndpoint) > 0 {
-		errs := validation.IsDNS1123Subdomain(cfg.API.ControlPlaneEndpoint)
+		parts := strings.FieldsFunc(cfg.API.ControlPlaneEndpoint, func(c rune) bool {
```

Comment:
I'm pretty certain there are other parsing utils that should be used here. 

/cc @kad

Reply:
I've modified this to use net.SplitHostPort().